### PR TITLE
Fixing code style automagically via php-cs-fixer by pre-commit hook

### PR DESCRIPTION
**Pull request by PatchRanger**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | -
| Related developer documentation PR URL | -
| Issues addressed (#s or URLs) | -
| BC breaks? | n
| Deprecations? | n

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds pre-commit git hook, which is installed/updated locally after each composer install/update.
Pre-commit git hook runs php-cs-fixer to fix code style before commit.
It will help to avoid code-style-related issues during development of Mautic. It will make every contributor to follow Mautic contributor guide - automatically, without any user effort.

#### Steps to test this PR:
1. Apply PR.
2. Run `composer update`.
3. Try to commit something: code style of the commit will be automatically fixed.

### CODE REVIEW

**Comment by dongilbert** [comment]
YES!!!

File: .php_cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+return Symfony\CS\Config\Config::create()
+    ->level(Symfony\CS\FixerInterface::SYMFONY_LEVEL)
+    ->fixers([
+        'short_array_syntax',
+        'ordered_use',
```
Comment (dongilbert) [inline .php_cs]: Please add `align_double_arrow`, and `align_equals` to this list.

Comment (PatchRanger) [inline .php_cs]: @dongilbert Thank you for feedback, done.

**Comment by dongilbert** [comment]
Please back this up to commit `91b9198`, and force push an update to your `PatchRanger-php-cs-fixer` branch. 

The timing for actually applying the full code style update hasn't yet been decided. Updating code style throughout the project makes most PR's have merge conflicts. We'll have to figure out the best timing before doing that.

**Comment by PatchRanger** [comment]
@dongilbert I understand, pushed forced to get rid of the previous commit.

**Comment by dongilbert** [comment]
@PatchRanger Thanks.

**Comment by PatchRanger** [comment]
@dongilbert You are welcome.
@escopecz Could you please clarify what kind of documentation is needed? This feature doesn't have any settings or configuration. It even doesn't need any setup effort - it  just begins to work automatically after next `composer update`. It has console notifications together with instructions for case if required library is not installed.
For me it looks mostly like "executed documentation": it just can't be used incorrectly - and therefore it (in my view) doesn't need any explicit documentation. That's why I need your clarification))

**Comment by escopecz** [comment]
I was thinking to mention it at https://github.com/mautic/mautic/blob/staging/.github/CONTRIBUTING.md.

**Comment by PatchRanger** [comment]
@dongilbert @escopecz Looks like it's now ready for merge, isn't it? I've enabled "Allow edits from maintainers" setting in order to let you make final changes.

**Comment by escopecz** [comment]
@PatchRanger tested and it works great! :+1: Thanks a million for this!

We have internal rules (we should make them public) with the new PHP dependencies:
1. Never run `composer update`. Only `composer install`
2. To comply with 1., the dependency must be installed by `composer require friendsofphp/php-cs-fixer` (add version if needed) and the PR must contain composer.lock with the changes.

This way the testers always run only `composer install` and won't install possibly unwanted updates of other libraries. Do you think you could update your PR that way?

**Comment by PatchRanger** [comment]
@escopecz I am glad you like it!))
1. I am not quite sure I understand why you avoid using `composer update` for `composer install`. As far as I know, they work the same manner given the same `composer.json`: so it doesn't matter whether you completely remove `vendor` folder and then run `composer install` or just run `composer update`. Just my 2 cents, I respect your way of doing and am going to make my PR to comply with it.
2. To make it working `composer require friendsofphp/php-cs-fixer` is not enough: git pre-commit hook must also be installed. It is done by `scripts` section of `composer.json` for both `composer install` and `composer update` - so I guess it still complies with your needs without any changes.

**Comment by alanhartless** [comment]
@PatchRanger We make it a habit not to run `composer update` so that we do not inadvertently update a dependency to an untested version and thus the reason we commit the composer.lock to the repository as well. Even though we've tried to lock down versions to patch only, we still want to control that process. The `composer.lock` file should be in sync with the `composer.json` file for new installations to simply run `composer install` to install tested and intended versions of the dependencies. 

But since this is a dev dependency, need to run `composer require --dev friendsofphp/php-cs-fixer` so that the `composer.lock` file is in sync - then commit the update to `composer.lock` as well. 

Thanks!

**Comment by PatchRanger** [comment]
@alanhartless Oh, now I see, thank you for clarification, never thought about it such way. I've run the command `composer require --dev friendsofphp/php-cs-fixer` and pushed all the changes it introduced.

**Comment by alanhartless** [comment]
Let's merge this last so we don't cause any conflicts with style changes - we'll run php-cs-fixer on the entire repository prior to release 2.2 so that all the code is then up to date. Then all the pending PRs for 2.2.1+ will have conflicts ;-)

**Comment by escopecz** [comment]
@PatchRanger sorry to bother you, but could you please merge this PR with latest staging? There is some conflict.

**Comment by PatchRanger** [comment]
@escopecz I don't mind) Conflict is gone.

**Pull request merged**